samples/star/src/main/kotlin/com/slack/circuit/star/home/HomePresenter.kt

```diff
@@ -141,79 +151,59 @@ class HomeUiFactory @Inject constructor() : Ui.Factory {
 
 private fun homeUi() = ui<HomeScreen.State> { state -> HomeContent(state) }
 
-@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun HomeContent(state: HomeScreen.State) {
   val systemUiController = rememberSystemUiController()
   systemUiController.setStatusBarColor(MaterialTheme.colorScheme.background)
   systemUiController.setNavigationBarColor(MaterialTheme.colorScheme.primaryContainer)
-  val modalState =
-    rememberModalBottomSheetState(
-      initialValue =
-        if (state.petListFilterState.showBottomSheet) ModalBottomSheetValue.Expanded
-        else ModalBottomSheetValue.Hidden
-    )
 
-  // Monitor bottom sheet state and emit event whenever the user dismisses the modal
-  LaunchedEffect(modalState) {
-    snapshotFlow { modalState.isVisible }
-      .collect { isVisible ->
-        // Toggle if state says the modal should be visible but the snapshot says it isn't.
-        if (state.petListFilterState.showBottomSheet && !isVisible)
-          state.eventSink(PetListFilterEvent(PetListFilterScreen.Event.ToggleAnimalFilter))
-      }
-  }
-
-  ModalBottomSheetLayout(
-    sheetState = modalState,
-    sheetContent = {
-      Column {
-        GenderFilterOption(state.petListFilterState, state.eventSink)
-        SizeFilterOption(state.petListFilterState, state.eventSink)
-      }
-    }
-  ) {
-    Scaffold(
-      modifier = Modifier.navigationBarsPadding().systemBarsPadding().fillMaxWidth(),
-      topBar = {
-        CenterAlignedTopAppBar(
-          title = {
-            Text("Adoptables", fontSize = 22.sp, color = MaterialTheme.colorScheme.onBackground)
-          },
-          colors =
-            TopAppBarDefaults.centerAlignedTopAppBarColors(
-              containerColor = MaterialTheme.colorScheme.background
-            ),
-          actions = {
-            IconButton(
-              onClick = {
-                state.eventSink(PetListFilterEvent(PetListFilterScreen.Event.ToggleAnimalFilter))
+  Scaffold(
+    modifier = Modifier.navigationBarsPadding().systemBarsPadding().fillMaxWidth(),
+    topBar = {
+      CenterAlignedTopAppBar(
+        title = {
+          Text("Adoptables", fontSize = 22.sp, color = MaterialTheme.colorScheme.onBackground)
+        },
+        colors =
+          TopAppBarDefaults.centerAlignedTopAppBarColors(
+            containerColor = MaterialTheme.colorScheme.background
+          ),
+        actions = {
+          val scope = rememberCoroutineScope()
+          val overlayHost = LocalOverlayHost.current
+          IconButton(
+            onClick = {
+              scope.launch {
+                val result = overlayHost.updateFilters(state.petListFilterState)
+                state.eventSink(
+                  PetListFilterEvent(PetListFilterScreen.Event.UpdatedFilters(result))
+                )
               }
```

Comment:
do you mean emit an event to show the bottom sheet as an aspect of state? If so – seems reasonable to me 👍. Would you mind if I do that in a quick followup since that's limited to the sample?

Reply:
Sure, a follow-up would be fine 👍